src/common/transformations/src/transformations/common_optimizations/eliminate_loop_inputs_outputs.cpp

```diff
@@ -0,0 +1,303 @@
+                        });
+                    }
+
+                    if (static_cast<int64_t>(result_idx) < body_condition_output_idx) {
+                        ++delta_body_condition_output_idx;
+                    }
+                    remove_result_inds.insert(result_idx);
+                }
+            }
+
+            if (body_param->get_output_target_inputs(0).size() == cur_remove_result_inds.size()) {
+                // remove Parameter node since all consumers (Result nodes) will be also removed
+                delete_inputs_outputs.emplace_front([=, &body_model]() {
+                    body_model->remove_parameter(body_param);
+                });
+                if (static_cast<int64_t>(input_description->m_body_parameter_index) < current_iteration_input_idx) {
+                    ++delta_current_iteration_input_idx;
+                }
+                // nothing to mark with input description
+                continue;
+            }
+
+            // move input description for unremoved node
+            if (const auto merged_input_desc = as_type_ptr<MergedD>(input_description)) {
+                if (cur_remove_result_inds.size() > 0) {
+                    process_inputs_outputs.emplace_back([=, &new_node]() {
+                        new_node->set_invariant_input(body_param, init_value);
+                    });
+                    continue;
+                }
+                const auto& body_res = body_results[merged_input_desc->m_body_value_index];
+                process_inputs_outputs.emplace_back([=, &new_node]() {
+                    new_node->set_merged_input(body_param, init_value, body_res);
+                });
+            } else if (const auto invariant_input_desc = as_type_ptr<InvariantD>(input_description)) {
+                process_inputs_outputs.emplace_back([=, &new_node]() {
+                    new_node->set_invariant_input(body_param, init_value);
+                });
+            } else if (const auto sliced_input_desc = as_type_ptr<SlicedD>(input_description)) {
+                process_inputs_outputs.emplace_back([=, &new_node]() {
+                    new_node->set_sliced_input(body_param,
+                                               init_value,
+                                               sliced_input_desc->m_start,
+                                               sliced_input_desc->m_stride,
+                                               sliced_input_desc->m_part_size,
+                                               sliced_input_desc->m_end,
+                                               sliced_input_desc->m_axis);
+                });
+            } else {
+                // unknown input description type
+                // this could only happen if new input description type was added after this transformation written
+                return false;
+            }
+        }
+
+        if (delete_inputs_outputs.size() == 0) {
```

Comment:
but we also replace Merged input descs with Invariants in this transformation.
do we have to stop the execution here even if replaced Merged to Invariant?

Reply:
discussed, looks fine.